src/main/java/net/pms/newgui/components/CustomJSpinner.java

```diff
@@ -0,0 +1,315 @@
+				((SpinnerIntModel) model).getStepSize()
+			));
+		}
+
+		if (enterMoveFocus) {
+			if (this.getEditor() instanceof CustomJSpinner.IntegerEditor) {
+				((CustomJSpinner.IntegerEditor) this.getEditor()).getTextField().addKeyListener(new KeyListener() {
+					@Override
+					public void keyTyped(KeyEvent e) {}
+					@Override
+					public void keyReleased(KeyEvent e) {}
+					@Override
+					public void keyPressed(KeyEvent e) {
+						if (e.getKeyCode() == KeyEvent.VK_ENTER) {
+							FocusManager.getCurrentManager().focusNextComponent();
+						}
+					}
+				});
+			}
+		}
+	}
+
+	protected JComponent createEditor(SpinnerModel model) {
+		if (model instanceof SpinnerDateModel) {
+			return new DateEditor(this);
+		} else if (model instanceof SpinnerNumberModel) {
+			return new NumberEditor(this);
+		} else if (model instanceof SpinnerListModel) {
+			return new ListEditor(this);
+		} else if (model instanceof SpinnerIntModel) {
+			return new IntegerEditor(this);
+		} else {
+			return new DefaultEditor(this);
+		}
+	}
+
+	public JToolTip createToolTip() {
+	    JToolTip tip = new HyperLinkToolTip();
+	    tip.setComponent(this);
+	    return tip;
+	}
+
+	protected class MouseWheelRoll  implements MouseWheelListener {
+
+		private int minimum, maximum, stepSize;
+		private CustomJSpinner spinner;
+
+		private MouseWheelRoll(CustomJSpinner spinner, int minimum, int maximum, int stepSize) {
+			this.spinner = spinner;
+			this.minimum = minimum;
+			this.maximum = maximum;
+			this.stepSize = stepSize;
+		}
+
+		@Override
+		public void mouseWheelMoved(MouseWheelEvent e) {
+	        if (e.getScrollType() != MouseWheelEvent.WHEEL_UNIT_SCROLL) {
+	            return;
+	        }
+
+	        int value = (int) spinner.getValue();
+	        value -= e.getWheelRotation() * stepSize;
+	        if (value == minimum + stepSize && minimum % stepSize != 0) {
+	        	value = ((minimum / stepSize) * stepSize) + stepSize;
+	        } else if (value == maximum - stepSize && maximum % stepSize != 0) {
+	        	value = (maximum / stepSize) * stepSize;
+	        }
+			value = Math.min(Math.max(value, minimum),maximum);
+	        spinner.setValue(value);
+		}
+	}
+
+    /**
+     * An editor for a <code>JSpinner</code> whose model is a
+     * <code>SpinnerIntModel</code>.  The value of the editor is
+     * displayed with a <code>JFormattedTextField</code> whose format
+     * is defined by a <code>NumberFormatter</code> instance whose
+     * <code>minimum</code> and <code>maximum</code> properties
+     * are mapped to the <code>SpinnerIntModel</code>.
+     */
+    public static class IntegerEditor extends DefaultEditor
+    {
+        // This is here until DecimalFormat gets a constructor that
+        // takes a Locale: 4923525
+        private static String getDefaultPattern(Locale locale) {
+            // Get the pattern for the default locale.
+            ResourceBundle rb = LocaleData.getNumberFormatData(locale);
```

Comment:
@Nadahar just posting for posterity, no need to respond til convenient :-):

compiling #646 with oracle jdk1.8.0_40:
```logtalk
Maven home: /f/git/apache-maven-3.2.1
Java version: 1.8.0_40, vendor: Oracle Corporation
Java home: /f/git/jdk1.8.0_40/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.2.0-4-amd64", arch: "amd64", family: "unix"
```
gives
```logtalk
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile) on project ums: Compilation failure: Compilation failure:
[ERROR] /f/git/UniversalMediaServer/src/main/java/net/pms/newgui/components/CustomJSpinner.java:[43,32] package sun.util.locale.provider does not exist
[ERROR] /f/git/UniversalMediaServer/src/main/java/net/pms/newgui/components/CustomJSpinner.java:[44,32] package sun.util.locale.provider does not exist
[ERROR] /f/git/UniversalMediaServer/src/main/java/net/pms/newgui/components/CustomJSpinner.java:[171,13] cannot find symbol
[ERROR] symbol:   class LocaleProviderAdapter
[ERROR] location: class net.pms.newgui.components.CustomJSpinner.IntegerEditor
[ERROR] /f/git/UniversalMediaServer/src/main/java/net/pms/newgui/components/CustomJSpinner.java:[172,23] cannot find symbol
[ERROR] symbol:   variable LocaleProviderAdapter
[ERROR] location: class net.pms.newgui.components.CustomJSpinner.IntegerEditor
[ERROR] /f/git/UniversalMediaServer/src/main/java/net/pms/newgui/components/CustomJSpinner.java:[174,13] cannot find symbol
[ERROR] symbol:   class LocaleResources
[ERROR] location: class net.pms.newgui.components.CustomJSpinner.IntegerEditor
[ERROR] /f/git/UniversalMediaServer/src/main/java/net/pms/newgui/components/CustomJSpinner.java:[176,22] cannot find symbol
[ERROR] symbol:   variable LocaleProviderAdapter
[ERROR] location: class net.pms.newgui.components.CustomJSpinner.IntegerEditor
```
and for another comparison compiling #616 with openjdk7 instead of my first oracle jdk (jdk1.7.0_05):
```logtalk
$ mvn -X compile
Apache Maven 3.2.1 (ea8b2b07643dbb1b84b6d16e1f08391b666bc1e9; 2014-02-14T12:37:52-05:00)
Maven home: /f/git/apache-maven-3.2.1
Java version: 1.7.0_79, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-7-openjdk-amd64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.2.0-4-amd64", arch: "amd64", family: "unix"
```
also gives the same error as my original test.

Applying the pom patch fixes it in both cases (though wierdly openjdk's ```rt.jar``` doesn't seem to contain a ```LocaleData.class```).